blockchain-source/src/cardano/point.rs

```diff
@@ -1,5 +1,22 @@
-#[derive(Debug, Clone, PartialEq, Eq)]
+use anyhow::anyhow;
+use dcspark_core::{BlockId, SlotNumber};
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
 pub enum Point {
     Origin,
-    BlockHeader { slot_nb: u64, hash: String },
+    BlockHeader { slot_nb: SlotNumber, hash: BlockId },
```

Comment:
this change will require to update carp as well, since this type is used there

Reply:
It doesn't require changing too much anyway in there though. It's only a couple of places. I can revert it and adjust the rest though, it's just a bit of a pity since we have the types for this here.

Reply:
I'm ok with your change, just wanted to point out this fact so we don't forget to update this in carp as well